### PR TITLE
[#148167] Add order for button

### PR DESCRIPTION
**Pull request by giladshanan**
# Release Notes

[Additional Order For Button](https://pm.tablexi.com/issues/148167)

It looks to me like it is possible to have a case where user cannot `:edit`, but they can `:switch_to`.  This is my attempt to handle both of those cases while also adding the new button inline with other buttons if they exist.

### CODE REVIEW

File: app/views/users/show.html.haml [hunk 1]
```diff
@@ -22,19 +22,17 @@
 
   = render_view_hook "additional_user_fields", f: f
 
-- if can? :edit, @user
-  %ul.inline
+%ul.inline
+  - if current_ability.can?(:switch_to, @user) && @user.active? && !current_facility.cross_facility?
+    %li= link_to text("search.results_table.order_for"), facility_user_switch_to_path(current_facility, @user), class: "btn btn-primary"
```
Comment (jhanggi) [inline app/views/users/show.html.haml:27]: I kind of feel like `btn-primary` is not the right styling. I'm not quite sure what is.

https://bootstrapdocs.com/v2.3.2/docs/base-css.html#buttons

Comment (jhanggi) [reply app/views/users/show.html.haml:27]: Also, can we put this after the "Edit" button?

Comment (jhanggi) [reply app/views/users/show.html.haml:27]: Maybe the edit button should just be a plain button? For the "edit" button for products we don't use `btn-primary`. What do you think?

File: app/views/users/show.html.haml [hunk 2]
```diff
@@ -22,19 +22,17 @@
 
   = render_view_hook "additional_user_fields", f: f
 
-- if can? :edit, @user
-  %ul.inline
+%ul.inline
+  - if current_ability.can?(:switch_to, @user) && @user.active? && !current_facility.cross_facility?
```
Comment (jhanggi) [inline app/views/users/show.html.haml:26]: We also have `Facility#single_facility?` which is the inverse of `cross_facility?`.

File: app/views/users/index.html.haml [hunk 3]
```diff
@@ -14,7 +14,6 @@
 
   - if @users.any?
     = render "search/results_table",
```
Comment (jhanggi) [inline app/views/users/index.html.haml:16]: Should we dig into this view and disable the order for link when you're in cross facility mode?

File: app/views/users/show.html.haml (hunk 1 above)
Comment (giladshanan) [reply app/views/users/show.html.haml:27]: Yup, that makes sense to me - edit button will just be `btn`, and order for will be `btn-primary`, and come after the edit button.

File: app/views/users/index.html.haml (hunk 3 above)
Comment (giladshanan) [reply app/views/users/index.html.haml:16]: I'm not sure - I thought Jeff said they were going to think on it.  Should be an easy change to make if the want it done.  Should I check in with them about it?

Comment (jhanggi) [reply app/views/users/index.html.haml:16]: You're right that it's probably an easy change so we can wait for them to give us the go ahead to change that.

File: app/views/users/show.html.haml [hunk 4]
```diff
@@ -22,19 +22,17 @@
...
       = link_to t("shared.activate"), facility_user_suspension_path(current_facility, @user), class: "btn", method: :delete
-  - if can?(:deactivate, @user) && !@user.suspended? && !@user.expired?
-    %fieldset.well
-      = simple_form_for @user, url: facility_user_suspension_path(current_facility, @user), method: :post do |f|
-        = f.input :suspension_note
-        = f.submit t("shared.suspend"), class: "btn btn-danger"
-
-
-
+- if can?(:edit, @user) && can?(:deactivate, @user) && @user.active?
```
[the remark this answers is not in the document]
Comment (giladshanan) [reply app/views/users/show.html.haml]: I was trying to replicate the existing logic.  I wrote a little script to help check permissions, and it looks like :edit and :deactivate always return the same (at least for the seed users).

**Pull request merged**